bee-tangle/src/tangle.rs

```diff
@@ -1,759 +1,133 @@
-        self.milestones.remove(&index);
-    }
-
-    /// Returns the milestone transaction corresponding to the given milestone `index`.
-    pub fn get_milestone(&'static self, index: MilestoneIndex) -> Option<TransactionRef> {
-        match self.get_milestone_hash(index) {
-            None => None,
-            Some(hash) => self.get_transaction(&hash),
+        match self.vertices.entry(hash) {
+            Entry::Occupied(_) => None,
+            Entry::Vacant(entry) => {
+                let vtx = Vertex::new(data, metadata);
+                let txref = vtx.get_data().clone();
+                entry.insert(vtx);
+                Some(txref)
+            }
         }
     }
 
-    /// Returns a [`VertexRef`] linked to the specified milestone, if it's available in the local Tangle.
-    pub fn get_latest_milestone(&'static self) -> Option<TransactionRef> {
-        todo!("get the last milestone index, get the transaction hash from it, and query the Tangle for it")
-    }
-
-    /// Returns the hash of a milestone.
-    pub fn get_milestone_hash(&'static self, index: MilestoneIndex) -> Option<Hash> {
-        match self.milestones.get(&index) {
-            None => None,
-            Some(v) => Some(*v),
+    #[inline]
+    fn add_child(&self, parent: TxHash, child: TxHash) {
+        match self.children.entry(parent) {
+            Entry::Occupied(mut entry) => {
+                let children = entry.get_mut();
+                children.push(child);
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(vec![parent]);
+            }
         }
     }
 
-    /// Returns whether the milestone index maps to a know milestone hash.
-    pub fn contains_milestone(&'static self, index: MilestoneIndex) -> bool {
-        self.milestones.contains_key(&index)
-    }
-
-    /// Retreives the solid milestone index.
-    pub fn get_solid_milestone_index(&'static self) -> MilestoneIndex {
-        self.solid_milestone_index.load(Ordering::Relaxed).into()
-    }
-
-    /// Updates the solid milestone index to `new_index`.
-    pub fn update_solid_milestone_index(&'static self, new_index: MilestoneIndex) {
-        self.solid_milestone_index.store(*new_index, Ordering::Relaxed);
-    }
-
-    /// Retreives the snapshot milestone index.
-    pub fn get_snapshot_milestone_index(&'static self) -> MilestoneIndex {
-        self.snapshot_milestone_index.load(Ordering::Relaxed).into()
-    }
-
-    /// Updates the snapshot milestone index to `new_index`.
-    pub fn update_snapshot_milestone_index(&'static self, new_index: MilestoneIndex) {
-        self.snapshot_milestone_index.store(*new_index, Ordering::Relaxed);
-    }
-
-    /// Retreives the last milestone index.
-    pub fn get_last_milestone_index(&'static self) -> MilestoneIndex {
-        self.last_milestone_index.load(Ordering::Relaxed).into()
+    /// Get the data of a vertex associated with the given `hash`.
+    pub fn get_data(&self, hash: &TxHash) -> Option<TxRef> {
```

Comment:
This has to be a transaction. I would then just call it `get`. Otherwise it's not consistent with `insert` or `contains`.

Reply:
Yea, I figured you would complain about that. My inner Thibault was already evil laughing at me.